api/controllers/CommunityController.js

```diff
@@ -55,6 +61,35 @@ module.exports = {
     .catch(res.serverError)
   },
 
+  addSlack: function (req, res) {
+    var code = req.query.code,
+      redirect_uri = req.protocol + '://' + process.env.DOMAIN + req.path,
+      options = {
+        uri: slackAuthAccess,
+        form: {
+          client_id: process.env.SLACK_APP_CLIENT_ID,
+          client_secret: process.env.SLACK_APP_CLIENT_SECRET,
+          code: code,
+          redirect_uri: redirect_uri
+        }
+      }
+
+    post(options).spread((resp, body) => {
+      var parsed = JSON.parse(body);
+      Community.find(req.param('communityId')).then(function (community) {
+        community.save({
+          slack_hook_url: parsed.incoming_webhook.url,
+          slack_team: parsed.team_name,
+          slack_configure_url: parsed.incoming_webhook.configuration_url
+        }, {patch: true})
+        .then(() => res.redirect(Frontend.Route.community(community) + '/settings?slack=1'))
+        .catch(() => res.redirect(Frontend.Route.community(community) + '/settings?slack=0'))
+      })
+    }).catch(err => {
+      res.redirect(Frontend.Route.community(community) + '/settings?slack=0')
```

Comment:
`community` is out of scope and so not defined in the catch handler
also, Standard JS says if you declare an err you have to handle it, so should be
`catch(() => {`
instead